python/tvm/micro/testing/pytest_plugin.py

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=invalid-name,redefined-outer-name
+""" microTVM testing fixtures used to deduce testing argument
+    values from testing parameters """
+
+import pathlib
+import os
+import datetime
+import pytest
+
+from tvm.contrib.utils import tempdir
+
+from .utils import get_supported_boards
+
+
+def pytest_addoption(parser):
+    """Adds more pytest arguments"""
+    parser.addoption(
+        "--board",
+        required=True,
+        choices=list(get_supported_boards("zephyr").keys())
+        + list(get_supported_boards("arduino").keys()),
+        help=(
+            "microTVM boards for tests. Board refers to instances"
+            "of microcontrollers/emulators defined in a platform."
+        ),
+    )
+    parser.addoption(
+        "--test-build-only",
+        action="store_true",
+        help="Only run tests that don't require physical hardware.",
+    )
+    parser.addoption(
```

Comment:
we should probably either change this to `--microtvm-debug` or move it upwards. could start with `--microtvm-debug` for now?

Reply:
done

Reply:
Big fan of this change!